provider/pkg/gen/compatibleTokens.go

```diff
@@ -0,0 +1,84 @@
+package gen
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pulumi/pulumi-azure-native/v2/provider/pkg/openapi"
+	"github.com/pulumi/pulumi-azure-native/v2/provider/pkg/openapi/paths"
+	"github.com/pulumi/pulumi-azure-native/v2/provider/pkg/util"
+)
+
+type Token string
+type ModuleLowered string
+type ResourceLowered string
+type NormalizedPath string
+
+type CompatibleTokensLookup struct {
+	byLoweredModuleResourceName map[ModuleLowered]map[ResourceLowered]map[Token]struct{}
+	byNormalizedPath            map[NormalizedPath]map[Token]struct{}
+}
+
+// NewCompatibleTokensLookup creates a new CompatibleResourceLookup from the given Azure modules and default versions.
+func NewCompatibleTokensLookup(tokenPaths map[string]string) (*CompatibleTokensLookup, error) {
+	lookup := CompatibleTokensLookup{
+		byLoweredModuleResourceName: map[ModuleLowered]map[ResourceLowered]map[Token]struct{}{},
+		byNormalizedPath:            map[NormalizedPath]map[Token]struct{}{},
+	}
+	if tokenPaths == nil {
+		tokenPaths = map[string]string{}
```

Comment:
We're not writing to this map, are we? So nil should be ok. If `util.MapOrdered` can't handle nil, it should be fixed, since nil maps are normal in Go.

Reply:
I _think_ that's true, but actually, we can just do this instead:

```suggestion
		return &lookup, nil
```